subxt/src/tx/tx_client.rs

```diff
@@ -465,4 +467,35 @@ where
         let dry_run_bytes = self.client.rpc().dry_run(self.encoded(), at).await?;
         dry_run_bytes.into_dry_run_result(&self.client.metadata())
     }
+
+    /// This returns an estimate for what the extrinsic is expected to cost to execute, less any tips.
+    /// The actual amount paid can vary from block to block based on node traffic and other factors.
+    pub async fn partial_fee_estimate(&self) -> Result<u128, Error> {
+        /// adapted from original type <https://paritytech.github.io/substrate/master/pallet_transaction_payment_rpc_runtime_api/struct.RuntimeDispatchInfo.html>
+        #[derive(Debug, Encode, Decode, Eq, PartialEq, Copy, Clone)]
+        pub struct RuntimeDispatchInfo {
+            /// Weight of this dispatch. Adapted from original type <https://paritytech.github.io/substrate/master/frame_support/weights/struct.Weight.html>
+            #[codec(compact)]
+            weight_ref_time: u64,
+            #[codec(compact)]
+            weight_proof_size: u64,
+            /// Class of this dispatch; an enum variant index. Adapted from original type <https://paritytech.github.io/substrate/master/frame_support/dispatch/enum.DispatchClass.html>
+            class: u8,
+            /// The inclusion fee of this dispatch; what we want:
+            partial_fee: u128,
+        }
+
+        let len_bytes: [u8; 4] = (self.encoded.0.len() as u32).to_le_bytes();
```

Comment:
I wonder what this interface expects here in terms of length; often I would expect an encoded `Compact<u32>` to be expected. I'll see if I can have a dig around and find out!

Reply:
Ok I finally dug around and found the answer, and I can see now that a u32 is the second argument for the runtime API so this is all ok.

In terms of code, a slight simplification would be:

```rust
// The first param is the encoded extrinsic:
let mut params = self.encoded().to_vec();
// The second param is the u32 length of said extrinsic:
self.encoded().len().encode_to(&mut params);

// .. pass `params` to the call
```